edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -60,43 +60,62 @@ def get_jwt_claim_mergeable_attributes(self):
         return get_setting('JWT_PAYLOAD_MERGEABLE_USER_ATTRIBUTES')
 
     def authenticate(self, request):
+        is_jwt_from_authorization_header = self._is_jwt_from_authorization_header(request)
```

Comment:
**Note for future:** This code would be much simpler and more clear if we simply check if the JWT Cookie is available, and if so, proceed as if it was used for authentication, even though it might have been the header. That means, if we _may_ be authenticating using a JWT cookie, then:
1. Use CSRF protection.
2. Respond with None in the event of failure to try other authentication classes.